include/gdrapi.h

```diff
@@ -122,7 +122,12 @@ int gdr_unmap(gdr_t g, gdr_mh_t handle, void *va, size_t size);
 
 // map_d_ptr is the user-space virtual address belonging to a mapping of a device memory buffer,
 // i.e. one returned by gdr_map()
+//
+// WARNING: The visibility of data to pre-launched kernels might be
```

Comment:
I would say "both integrity and ordering of data as observed by pre-launched GPU work is not guaranteed by this API. For more..."